content/en/docs/How To/create_input.md

```diff
@@ -0,0 +1,102 @@
+---
+title: "Create a custom input component"
+date: 2020-11-02
+weight: 650
+---
+The input component goal is to retrieve Records in form of bytes and send them to the filterchain.
+
+The input isn't in charge of splitting/parsing the input data into Records (that is done by Baker),
+but only retrieving them as fast as possible in raw format adding, if any, metadata to them.
+
+The input shares a [`*Data`](https://pkg.go.dev/github.com/AdRoll/baker#Data) channel with the
+filter chain. The channel size is customizable with `[input] chansize=<value>` (default to 1024).
+
+To create an input and make it available to Baker, one must:
+
+* Implement the [Input](https://pkg.go.dev/github.com/AdRoll/baker#Input) interface
+* Add an [`InputDesc`](https://pkg.go.dev/github.com/AdRoll/baker#InputDesc) for the input to
+the available inputs in [`Components`](https://pkg.go.dev/github.com/AdRoll/baker#Components)
+
+## Daemon vs Batch
+
+The input component determines the Baker behavior between a batch processor or a long-living daemon.
+
+If the input exits when its data processing has completed, then Baker waits the topology to end
+and then exits.
+
+If the input never exits, then Baker acts as a daemon.
+
+## Data
+
+The [`Data`](https://pkg.go.dev/github.com/AdRoll/baker#Data) object that the input must fill in
+with read data has two fields: `Bytes` that must contain the raw read bytes (possible contains
+more records) and `Meta`.
+
+[`Metadata`](https://pkg.go.dev/github.com/AdRoll/baker#Metadata) can contain any information
+that the input wants to share with the other components of the pipeline about the chunk of bytes
+produced.  
+Typical information could be the timestamp of the read, details about the source, etc. 
+
+Those metadata will be associated to each record extracted from `Bytes` and will be available
+to the whole pipeline.
```

Comment:
```suggestion
[`Metadata`](https://pkg.go.dev/github.com/AdRoll/baker#Metadata) can contain additional 
information Baker will associate with each of the serialized Record contains in `Data`.  
Typical information could be the time of retrieval, the filename (in case `Records` come a file), etc.
```